errornotifier/airbraker.go

```diff
@@ -6,22 +6,31 @@ package errornotifier
 import (
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
 	"runtime/debug"
 
+	"github.com/airbrake/gobrake"
 	"github.com/theplant/appkit/log"
-	gobrake "gopkg.in/airbrake/gobrake.v1"
 )
 
 // Notifier defines an interface for reporting error
 type Notifier interface {
-	Notify(interface{}, *http.Request) error
+	Notify(err interface{}, r *http.Request, context map[string]interface{})
 }
 
 // airbrakeNotifier is a Notifier that sends messages to Airbrake,
 // constructed via NewAirbrakeNotifier
 type airbrakeNotifier struct {
-	*gobrake.Notifier
+	notifier *gobrake.Notifier
+}
+
+func (n *airbrakeNotifier) Notify(e interface{}, req *http.Request, context map[string]interface{}) {
+	notice := n.notifier.Notice(e, req, 1)
```

Comment:
This is how the correct stack trace is captured, right?

Reply:
right, copy from the official package: https://github.com/airbrake/gobrake/blob/7af14bfb7d8ff856c1fd7df4e0dd83caba91d170/notifier.go#L219, and I test is ok.